common/components/WalletDecrypt/LedgerNano.jsx

```diff
@@ -1,26 +1,145 @@
+// @flow
+import './LedgerNano.scss';
 import React, { Component } from 'react';
 import translate from 'translations';
+import DeterministicWalletsModal from './DeterministicWalletsModal';
+import LedgerWallet from 'libs/wallet/ledger';
+import Ledger3 from 'vendor/ledger3';
+import LedgerEth from 'vendor/ledger-eth';
+import u2f from 'vendor/u2f-api';
+import DPATHS from 'config/dpaths.js';
+
+const DEFAULT_PATH = DPATHS.LEDGER[0].value;
+
+type State = {
+  publicKey: string,
+  chainCode: string,
+  dPath: string,
+  error: ?string,
+  isLoading: boolean
+};
 
 export default class LedgerNanoSDecrypt extends Component {
+  props: { onUnlock: any => void };
+  state: State = {
+    publicKey: '',
+    chainCode: '',
+    dPath: DEFAULT_PATH,
+    error: null,
+    isLoading: false
+  };
+
+  _handlePathChange = (dPath: string) => {
+    this.setState({ dPath });
+    this._handleConnect(dPath);
+  };
+
+  _handleConnect = (dPath: string = this.state.dPath) => {
+    this.setState({
+      isLoading: true,
+      error: null
+    });
+
+    const ledger = new Ledger3('w0w');
+    const ethApp = new LedgerEth(ledger);
+
+    ethApp.getAddress(
+      dPath,
+      (res, err) => {
+        if (err) {
+          err = err.errorCode ? u2f.getErrorByCode(err.errorCode) : err;
```

Comment:
May through this piece of logic into a library function. It's used a few times throughout the PR.

Reply:
Yep, digital bitbox looked similar, though it also had a DigitalBitboxEth.getError function. I added a static function to the DigitalBitbox class to the tune of decodeError or something like that.